pkg/trace/traceutil/span.go

```diff
@@ -13,13 +13,22 @@ const (
 
 	// measuredKey is a special metric flag that marks a span for trace metrics calculation.
 	measuredKey = "_dd.measured"
+	// tracerTopLevelKey is a metric flag set by tracers on top_level spans
+	tracerTopLevelKey = "_dd.top_level"
 )
 
 // HasTopLevel returns true if span is top-level.
 func HasTopLevel(s *pb.Span) bool {
 	return s.Metrics[topLevelKey] == 1
 }
 
+// UpdateTracerTopLevel sets _top_level tag on spans flagged by the tracer
+func UpdateTracerTopLevel(s *pb.Span) {
+	if s.Metrics[tracerTopLevelKey] == 1 {
+		SetMetric(s, topLevelKey, 1)
```

Comment:
What's the use of having both topLevelKey and  tracerTopLevelKey set?

Reply:
It's to keep track for now of when it's set by the tracer

Reply:
Seems a bit wasteful but we can change it later.

Reply:
Can we just have this code inside `agent.go`. I don't think this warrants a function in this package.